lib/child-writer/writer-proxy.js

```diff
@@ -109,62 +122,77 @@ permissions.isElevated().then((elevated) => {
   console.log('Re-spawning with elevation')
 
   return new Bluebird((resolve, reject) => {
-    ipc.config.id = process.env.IPC_CLIENT_ID
-    ipc.config.silent = true
 
-    // > If set to 0, the client will NOT try to reconnect.
-    // See https://github.com/RIAEvangelist/node-ipc/
-    //
-    // The purpose behind this change is for this process
-    // to emit a "disconnect" event as soon as the GUI
-    // process is closed, so we can kill the CLI as well.
-    ipc.config.stopRetrying = 0
+    let child = null
+
+    /**
+     * @summary Emit an object message to the IPC server
+     * @function
+     * @private
+     *
+     * @param {Buffer} data - json message data
+     *
+     * @example
+     * emitMessage(Buffer.from(JSON.stringify({
+     *   foo: 'bar'
+     * })));
+     */
+    const emitMessage = (data) => {
+      // Output from stdout/stderr coming from the CLI might be buffered,
+      // causing several progress lines to come up at once as single message.
```

Comment:
Could the buffering mentioned here mean that we might also get incomplete lines? e.g.
```
oo"}\n{"id":"bar"}\n{"id":"baz"}
```
or
```
{"id":"foo"}\n{"id":"bar"}\n{"id":"baz"
```
AFAICT `utils.splitObjectLines` has no handling for that.

Or are we sure we'll always get some number of "complete" lines?

Reply:
We don't write partial JSON to stdio, so it'll always be multiple lines containing valid, parseable JSON; btw this is also not new code, just moved it a bit.

Reply:
> We don't write partial JSON to stdio

I wondered if maybe it was possible that whatever "buffering" is being used is getting full, and so flushing before we expect it to?

> btw this is also not new code, just moved it a bit.

Yer I know!  I was just wondering if this might be another possible corner-case that we're not dealing with ;)

Reply:
> I wondered if maybe it was possible that whatever "buffering" is being used is getting full, and so flushing before we expect it to?

The `console.log()`ing flushes it, so the buffering in terms of partials won't be an issue.

> Yer I know! I was just wondering if this might be another possible corner-case that we're not dealing with ;)

Yeah, was wondering the same while I was reading over it, but then realized that we're controlling that, so didn't worry about it anymore. Still not worried. Also think that we definitely would've seen it, should it have happened.